portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +30,54 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private List<String> messages = new ArrayList<>();
+  private String json;
+
+  /** Converts Messages ArrayList to Json */
+  public void toJson() throws IOException {
+    json = new Gson().toJson(messages);
+  }
+
+  /** Redirection to the main page */
+  private void doRedirect(HttpServletResponse response) throws IOException {
+    response.sendRedirect(
+      "/../../../../../../index.html"
```

Comment:
```suggestion
      "~/portfolio/src/main/webapp/index.html
```
You can use a path that is relative to the home directory so that this is easier to read. Also what I've suggested may be more resilient to change, since the top-level file structure likely will stay the same.

Reply:
Got it! Thank you Aki!